operator/pkg/lib/sidecars/restart/rollout_action.go

```diff
@@ -0,0 +1,63 @@
+package restart
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const rolloutTimeoutMessage = "pod could not be rolled out by resource owner's controller."
+
+const annotationName = "kubectl.kubernetes.io/restartedAt"
+
+func newRolloutAction(object actionObject) restartAction {
+	return restartAction{
+		run:    rolloutRun,
+		object: object,
+	}
+}
+
+func rolloutRun(ctx context.Context, k8sclient client.Client, object actionObject) ([]RestartWarning, error) {
+	var obj client.Object
+	var err error
+
+	switch object.Kind {
+	case "DaemonSet":
+		obj = &appsv1.DaemonSet{}
+	case "Deployment":
+		obj = &appsv1.Deployment{}
+	case "ReplicaSet":
+		obj = &appsv1.ReplicaSet{}
+	case "StatefulSet":
+		obj = &appsv1.StatefulSet{}
+	default:
+		return nil, fmt.Errorf("kind %s not found", object.Kind)
```

Comment:
Maybe "not found" can be misleading. I think "not supported" gives a better idea that an implementation might be missing or that a wrong kind reached this code.